File: Java/Spark/SparkSample_JAVA/src/main/java/com/spark_java/sample/Transform_Union.java
package com.spark_java.sample;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class Transform_Union {

	public static void main(String[] args) throws Exception {
		
		
		SparkConf sparkConf = new SparkConf().setAppName("Transform_FlatMap").setMaster("local[*]");
		JavaSparkContext sc = new JavaSparkContext(sparkConf);

		JavaRDD<Integer> rdd = sc.parallelize(Arrays.asList(1, 2, 3, 4, 5, 6));
		JavaRDD<Integer> a = rdd.sample(false,  0.5, 1);
		JavaRDD<Integer> b = rdd.sample(true,  0.5, 1);
		List<Integer> c = a.union(b).collect();
		
		rdd.foreach(s -> System.out.println(s));
		
		System.out.println("Result of A ----------------------------------");
		a.foreach(s -> System.out.println(s));
		
		System.out.println("Result of B ----------------------------------");
		b.foreach(s -> System.out.println(s));
		
		System.out.println("Result of Union ----------------------------------");
		c.forEach(s -> System.out.println(s));
		
		sc.stop();		
		sc.close();
		
	}
}

File: Java/Spark/spark-edu/src/main/scala/com/bistel/edu/TransformUnion.scala
package com.bistel.edu

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext;
import org.apache.spark.SparkContext._;

object TransformUnion extends App {
  
  val conf = new SparkConf().setAppName("TransformUnion").setMaster("local[*]")
  val sc = new SparkContext(conf)
  
  val rdd = sc.parallelize(1 to 3)
  val a = rdd.sample(false, 1, 1)
  val b = rdd.sample(true, 1, 1)
  val c = a.union(b).collect
  //(a++b).collect
  
  println("A============================")
  a.foreach(println)
  
  println("B============================")
  b.foreach(println)
  
  println("C============================")
  c.foreach(println)
}